### PR TITLE
[r-mr1] radio: Allow finding qti uce service

### PR DESCRIPTION
**Pull request by ix5**
### radio: Allow finding qti.uceservice

`qcrild` as part of the radio domain needs to find `com.qualcomm.qti.uceservice::IUceService`

### Move public definitions back to vendor
    
There was no need to keep the definitions shared between system and vendor when the domains using them were contained in our vendor sepolicy only.

### Revert shell dmesg
Original commit was typo'd, it meant to allow `shell`, not `system_server` to read syslog.
However, that'd be a neverallow:
`system/sepolicy/domain.te:519`:
```
neverallow appdomain kernel:system { syslog_read syslog_mod syslog_console };
```
`system/sepolicy/private/shell.te:22`:
```
.# XXX Transition into its own domain?
app_domain(shell)
```

### CODE REVIEW

**Review by MarijnS95 — CHANGES_REQUESTED** [review]
I have a hunch this should be:

```
hal_attribute_hwservice(hal_telephony, vnd_qti_uce_hwservice)
```

In `hal_telephony.te`.

**Comment by ix5** [comment]
@MarijnS95 might taking another look?

**Comment by MarijnS95** [comment]
@ix5 That's pretty crazy, shouldn't `ims` be a _`server`_ domain to be allowed to _add_ (host) `IUCEService`? Or was that wrong judgement by 346be220646c1feb5a533cff05d53c082927ee69?

If it is, please:
```
Fixes: 346be22 ("ims: Label new services and grant add_hwservice access.")
```

But I don't see any other domain with the rights to find/host this service... Perhaps because I have never used this and anyone doing so (PE) is still in permissive.

- Who hosts this service?
- Do we at some point need a new hwservice "attribute" to shove ims-specific services under?

File: vendor/ims.te [hunk 1]
```diff
@@ -7,7 +7,8 @@ net_domain(ims)
 add_hwservice(ims, vnd_qti_ims_callinfo_hwservice)
 add_hwservice(ims, vnd_qti_ims_factory_hwservice)
 add_hwservice(ims, vnd_qti_imscms_hwservice)
-add_hwservice(ims, vnd_qti_uce_hwservice)
+
+hal_client_domain(ims, hal_telephony)
```
Comment (MarijnS95) [inline vendor/ims.te:11]:
```suggestion
hal_server_domain(ims, hal_telephony)
```

**Review by MarijnS95 — CHANGES_REQUESTED** [review]
```console
akatsuki:/ # lshal | grep -i uce
X     Y android.hidl.base@1.0::IBase/com.qualcomm.qti.uceservice                                  0/1        770    518
DM    Y com.qualcomm.qti.uceservice@2.0::IUceService/com.qualcomm.qti.uceservice                  0/1        770    518
DM    Y com.qualcomm.qti.uceservice@2.1::IUceService/com.qualcomm.qti.uceservice                  0/1        770    518
akatsuki:/ # ps -A 770
USER           PID  PPID     VSZ    RSS WCHAN            ADDR S NAME
system         770     1   59588   4536 binder_io+          0 S imsrcsd
```
So `imsrcsd` hosts it, which is `ims_exec`, so `ims`. Should be a server domain, did you run-test this on-device?

**Comment by MarijnS95** [comment]
We should get @paulbouchara to test this in parallel with us on PE since they use IMS and go enforcing soon™, in particular the remaining `add_hwservice`s should probably be moved to `hal_attribute_hwservice` while falling under the `hal_server_domain(ims, hal_telephony)` catch-all.

Iirc `hwbinder_use` at the bottom of `ims.te` is granted by one of these macros as well?

**Comment by Paulbouchara** [comment]
> We should get @Paulbouchara to test this in parallel with us on PE since they use IMS and go enforcing soon™, in particular the remaining `add_hwservice`s should probably be moved to `hal_attribute_hwservice` while falling under the `hal_server_domain(ims, hal_telephony)` catch-all.
> 
> Iirc `hwbinder_use` at the bottom of `ims.te` is granted by one of these macros as well?

The next PE release will be on enforcing, as every releases that will follow, so sure, I'll test this asap when I'll be back and report with logs

File: vendor/ims.te (hunk 1 above)
Comment (ix5) [reply vendor/ims.te:11]: You are correct, `imsrcsd` should also be a server domain for hal_telephony.

However, now `ims`, `rild` and `cnd` are all server domains. So much for separation of concerns, huh...

Comment (MarijnS95) [reply vendor/ims.te]: @ix5 Corrrect. Just hacked together an idea for separating `ims` out again, but remains to be seen how many ties it has into the `radio` domain (qcrild)... https://github.com/MarijnS95/device-sony-sepolicy/compare/ims-attribute-group

**Comment by MarijnS95** [comment]
@ix5 One more thing: the denial shown in the logs tries to find (and perhaps add later) this service for the `radio` domain, but this PR allows hosting (adding) it from the `ims` domain...

**Comment by ix5** [comment]
@MarijnS95 `ims` was already hosting the service via the [`add_hwservice()` macro](https://android.googlesource.com/platform/system/sepolicy/+/refs/tags/android-11.0.0_r33/public/te_macros#639).

That macro, however, forbids any other domains (even parent ones) from adding the service. So `ims` was not hosting this service in a server capacity, but just on its own.

The only way for a coredomain such as `radio` to access the IUceService is via `hal_` server/client attribute empowerment.

As such, we need to keep the status quo (ims still hosts the service), just change the language.

**Comment by MarijnS95** [comment]
Final verdict: `radio` only needs to find the service (and have binder communication with the domain hosting it, `ims`) so we're better off _not_ adding it to `hal_telephony` since then:

- `radio` is a server domain, and would incorrectly be allowed to host it;
- `ims` would become a server domain, and is incorrectly allowed to host all other services in the list.

As posted in https://github.com/sonyxperiadev/device-sony-sepolicy/compare/master...MarijnS95:ims-attribute-group we should check whether more hwservices in the `hal_telephony` domain are actually not hosted by `radio` at all, and are better separated so that `radio` can only _read_ them.

**Comment by ix5** [comment]
For posterity, this was the original description of the "solution" using HAL attributes we discarded:

---

This accomplishes the following:
- Wrestle control over `vnd_qti_uce_hwservice` from `ims` domain
  ([`add_hwservice()` macros](https://android.googlesource.com/platform/system/sepolicy/+/refs/tags/android-11.0.0_r33/public/te_macros#639) includes a `neverallow` for other  domains to add the service in any way)
- All clients of hal_telephony are now allowed to access `com.qualcomm.qti.uceservice::IUceService`
  This includes radio, which is a client domain of `hal_telephony`.

Since ims is no longer allowed access to vnd_qti_uce_hwservice, it needs to be converted into a server of hal_telephony via the `hal_server_domain()` macro.
N.b.: `imsrcsd` hosts the `IUceService`, so it really needs to be a server and not only a client.

Denial:
```
avc: denied  { find } for interface=com.qualcomm.qti.uceservice::IUceService \
    sid=u:object_r:radio:s0 scontext=u:object_r:radio:s0 \
    tcontext=u:object_r:vnd_qti_uce_hwservice:s0 tclass=hwservice_manager
```
---
Plus a small fixup

**Pull request merged**